Describe the view_component flow chart (#10)
rl-institut/open_plan: docs/view_components/flow_chart.rst

.. _flowchart-label:

Flow-chart
----------

> The flow chart is a map of all the sequential steps of the simulation in one view-component. The below sections present the proposed structure for this view-component.
> 
> The following are the advantages of the flow chart view-component:

Reviewer: ```suggestion
The flow chart is a view component that shows all the sequential steps of the tool in order to get results.

A flow chart view-component has a number of advantages:
```

* Helps in sequentially progressing through the modeling and simulation process
* Allows the user to jump between different steps of the simulation
* Brings a design consistency and uniformity to the UI of the tool

> 

Reviewer: ```suggestion
* Splits the tool in different views which increases the user friendliness 

```

Attributes
^^^^^^^^^^

> * Clickable, identical buttons

Reviewer: ```suggestion
* Clickable buttons that are identical throughout the different views
```

> * Ribbon shaped, clickable icons/buttons placed sequentially as per the simulation progression order.

Reviewer: would go more in the "rendering" part

Author: Done, moved them

* Each button is a step in the simulation process
* Initially, only the first step (or button) is highlighted in a certain color, which notifies the user that the view associated with that step has to be fulfilled (provided with values for the parameters in that view-component) before they can progress to further steps.
* Rest of the buttons are greyed out.

> * All the buttons for which the requirements of the associated simulation steps have been fulfilled turn green (or a different color).

Reviewer: I would avoid color specifications which seems to apply more to "rendering"

> * Initially, only the first step (or button) is highlighted in a certain color, which notifies the user that the view associated with that step has to be fulfilled (provided with values for the parameters in that view-component) before they can progress to further steps.
> * Rest of the buttons are greyed out.
> * All the buttons for which the requirements of the associated simulation steps have been fulfilled turn green (or a different color).

Reviewer: ```suggestion
* Each step has to be completed one at a time. Once a step has met its minimum requirements, the button of the next step becomes clickable. Therefore, the user can come back to the steps he has already seen but he cannot click a random step. 
```

Reviewer 2: This is not an attribute but a requirement

Actions
^^^^^^^

> * Clicking on the active (orange color) buttons alters the view to the corresponding view-component of the simulation. E.g.: Clicking on 'Project Setup' will alter to a view with the options to enter values for various project parameters.
> * Clicking on the greyed-out (inactive) buttons does not elicit any response.

Reviewer: ```suggestion
* To make the next step of the flowchart clickable, the user has to fulfill the requirements of the current step. Each step has various mandatory and optional fields.
* Clicking on the steps of the flowchart that have been fulfilled already allows the user to make modifications. 
* Clicking on the steps that are not clickable make a small text buble appear saying the previous step needs to be fulfilled first.
```

.. _flowchart-label:

> Flow-chart

Reviewer 2: Maybe we should call it "progression bar" or "progression-menu" instead of flow-chart ? What do you think @TheOneAndra, @mahendrark ?

Author: I agree with Progression bar

----------
The flow chart is a view component that shows all the sequential steps of the tool in order to get results.

A flow chart view-component has a number of advantages:

* Helps in sequentially progressing through the modeling and simulation process
* Allows the user to jump between different steps of the simulation
* Brings a design consistency and uniformity to the UI of the tool
* Splits the tool in different views which increases the user friendliness 

Attributes
^^^^^^^^^^

* Clickable buttons that are identical throughout the different views
* Ribbon shaped, clickable icons/buttons placed sequentially as per the simulation progression order.
* Each button is a step in the simulation process
* Each step has to be completed one at a time. Once a step has met its minimum requirements, the button of the next step becomes clickable. Therefore, the user can come back to the steps he has already seen but he cannot click a random step. 

> 

Reviewer 2: These are not attributes: attributes for flow chart are
- ordered list of steps
- index of current step

The fact that the steps can be clicked on is defined within "Actions", the fact that the steps are button or links is described under Rendering of the view-component

Actions
^^^^^^^

* To make the next step of the flowchart clickable, the user has to fulfill the requirements of the current step. Each step has various mandatory and optional fields.
* Clicking on the steps of the flowchart that have been fulfilled already allows the user to make modifications. 
* Clicking on the steps that are not clickable make a small text buble appear saying the previous step needs to be fulfilled first.

Requirement
^^^^^^^^^^^

Link with other view-components
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Rendering of the view-component
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> .. image:: docs/assets/flow_chart.png

Reviewer 2: this is only one example of rendering, I would prefer that we describe rendering with words if possible. For example the rendering could contain proposals for what steps should be in our component-view.

.. _flowchart-label:

Flow-chart
----------
The flow chart is a view component that shows all the sequential steps of the tool in order to get results.

A flow chart view-component has a number of advantages:

* Helps in sequentially progressing through the modeling and simulation process
* Allows the user to jump between different steps of the simulation
* Brings a design consistency and uniformity to the UI of the tool
* Splits the tool in different views which increases the user friendliness 

Attributes
^^^^^^^^^^

> * Each button is a step in the simulation process

Reviewer 2: If you answer the question "what are the attributes which characterize the view-component" (not what they do or how can I interact with them) you get what should be listed under attributes. 

Here you say that a button is a step before defining steps. Besides, the fact that it should be a button is something that I would propose in the rendering part: there is a list of steps (attributes) + user can click on step (action) = steps can be button (rendering).

* The list of steps in the progression bar are as follows:
    #. Project Setup
    #. Inputs
    #. System Inputs
    #. Visualization
    #. Constraints
    #. Simulations
    #. Results
* Each step has an associated index, with the progression being bar considered as an array

> * The current state of the view depends on the index of the current step

Reviewer 2: This is more an action/requirement

Author: I would keep it as an attribute because the view could have been the default where there was no requirement for the user to have done something to reach that state.

> * Each step has to be completed one at a time. Once a step has met its minimum requirements, the button of the next step becomes clickable. Therefore, the user can come back to the steps he has already seen but he cannot click a random step. 

Reviewer 2: this is a requirement and not an attribute
